tools/Dashboard/DevHome.Dashboard/Views/DashboardView.xaml.cs

```diff
@@ -346,10 +352,28 @@ private async void RestorePinnedWidgets(object sender, RoutedEventArgs e)
                     if (!string.IsNullOrEmpty(stateStr))
                     {
                         var stateObj = System.Text.Json.JsonSerializer.Deserialize(stateStr, SourceGenerationContext.Default.WidgetCustomState);
+
                         if (stateObj.Host == WidgetHelpers.DevHomeHostName)
                         {
-                            var size = await widget.GetSizeAsync();
-                            AddWidgetToPinnedWidgetsAsync(widget, size);
+                            var position = stateObj.Position;
+                            if (position >= 0)
+                            {
+                                if (!restoredWidgetsWithPosition.ContainsKey(position))

```

Comment:
You can branch on `!restoredWidgetsWithPosition.TryAdd(position, widget)` to remove a double lookup 🙂